LocationUpdatesBackgroundKotlin/app/src/main/java/com/google/android/gms/location/sample/locationupdatesbackgroundkotlin/data/MyLocationManager.kt

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright (C) 2020 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.data
+
+import android.Manifest
+import android.app.PendingIntent
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import androidx.lifecycle.MutableLiveData
+import com.google.android.gms.location.FusedLocationProviderClient
+import com.google.android.gms.location.LocationRequest
+import com.google.android.gms.location.LocationServices
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.LocationUpdatesBroadcastReceiver
+import com.google.android.gms.location.sample.locationupdatesbackgroundkotlin.hasPermission
+import java.util.concurrent.TimeUnit
+
+private const val TAG = "MyLocationManager"
+
+/**
+ * Manages all location related tasks for the app.
+ */
+class MyLocationManager private constructor(private val context: Context) {
+
+    val trackingLocation: MutableLiveData<Boolean> by lazy {
```

Comment:
do you need it to be `byLazy`? Why no just instantiate it here?

Also, by exposing a `MutableLiveData` it means that any class that has access to it will be able to set the value. I would rather use a backing field and expose a `LiveData`

Reply:
Done.